server/resource_server/repository/repository.go

```diff
@@ -324,3 +324,17 @@ func (r *Repository) IsBackendURIExists(backendURL string) (bool, error) {
 	}
 	return count > 0, nil
 }
+
+func (r *Repository) RegisterPrecheckUser(req dto.RegisterUserPrecheckDTO, salt string, iterCount int) (string, int, error) {
```

Comment:
No need to return salt and iterCount from this method, since these values are already known to the caller (since they are passed here as an input). We can just make this method return an ``error`` type which would reflect whether or not user's data was persisted successfully.